pkg/filter/mqttclientauth/mqttauth.go

```diff
@@ -48,20 +44,21 @@ type (
 		filterSpec *pipeline.FilterSpec
 		spec       *Spec
 		authMap    map[string]string
+		salt       string
 	}
 
-	// Spec is spec for MQTTClientAuth
-	// authFile format:
-	// - username: test
-	//   passBase64: dGVzdA==
+	// Spec is spec for MQTTClientAuth.
+	// For security of password, passwords in yaml file should be salted SHA256 checksum.
+	// password = sha256sum(connect.password + salt)
 	Spec struct {
-		AuthFile string `yaml:"authFile" jsonschema:"required"`
+		Salt string  `yaml:"salt" jsonschema:"omitempty"`
```

Comment:
is `Salt` a fixed value for all user/password pairs? then it is helpless for security and can be remvoed.

Reply:
it is fixed. I think it mainly used for solving rainbow table attack. making such table may need some time and user can update during that time. One salt for one password is also ok...

Reply:
I had a similar question about this, but I think @suchen-sci's explanation makes sense.

Reply:
Yes, got it, it could be some help though I think building a rainbow table won't take too much time.

Reply:
so one salt for one password? and also support salt for all passwords, let user to choose. If they want more secure use one salt for one passwrod?